.github/workflows/pitest.yml

```diff
@@ -7,673 +7,50 @@ on:
-        with:
-          name: pitest-tree-walker-coverage-report
-          path: staging
-          retention-days: 7
-
-  pitest-utils:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Setup local maven cache
-        uses: actions/cache@v2
+      - name: Download Groovy 4.0.3
+        uses: sdkman/sdkman-action@master
+        id: sdkman
         with:
-          path: ~/.m2/repository
-          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: pitest-utils
-        run: |
-          ./.ci/pitest.sh pitest-utils
-      - name: Stage results
-        if: failure() || github.ref == 'refs/heads/master'
+          candidate: groovy
+          version: 4.0.3
+      - name: Set up Groovy 4.0.3
         run: |
-          mkdir staging && cp -r target/pit-reports/ staging
-      - name: Archive code coverage results
-        if: failure() || github.ref == 'refs/heads/master'
-        uses: actions/upload-artifact@v2
-        with:
-          name: pitest-utils-coverage-report
-          path: staging
-          retention-days: 7
-
-  pitest-whitespace:
-    runs-on: ubuntu-latest
-    steps:
+          unzip ${{ steps.sdkman.outputs.file }} -d ${{ runner.temp }};
+          mv ${{ runner.temp }}/groovy-4.0.3 ${{ runner.temp }}/groovy
+          GROOVY_HOME=${{ runner.temp }}/groovy
+          echo "GROOVY_HOME=${{ runner.temp }}/groovy" >> $GITHUB_ENV
+          echo "$GROOVY_HOME/bin" >> $GITHUB_PATH
       - name: Setup local maven cache
         uses: actions/cache@v2
         with:
           path: ~/.m2/repository
           key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
       - name: Checkout
         uses: actions/checkout@v2
-      - name: pitest-whitespace
-        run: |
-          ./.ci/pitest.sh pitest-whitespace
-      - name: Stage results
-        if: failure() || github.ref == 'refs/heads/master'
+      - name: Generate ${{ matrix.profile }}
         run: |
-          mkdir staging && cp -r target/pit-reports/ staging
-      - name: Archive code coverage results
-        if: failure() || github.ref == 'refs/heads/master'
-        uses: actions/upload-artifact@v2
-        with:
-          name: pitest-whitespace-coverage-report
-          path: staging
-          retention-days: 7
-
-  pitest-xpath:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Setup local maven cache
-        uses: actions/cache@v2
-        with:
-          path: ~/.m2/repository
-          key: checkstyle-maven-cache-${{ hashFiles('**/pom.xml') }}
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: pitest-xpath
+          mvn -e --no-transfer-progress -P"${{ matrix.profile }}" clean test-compile \
+          org.pitest:pitest-maven:mutationCoverage
+      - name: Check ${{ matrix.profile }}
         run: |
-          ./.ci/pitest.sh pitest-xpath
+          groovy ./.ci/pitest.groovy ${{ matrix.profile }}
```

Comment:
ci/pitest-survival-check.groovy

Reply:
Renamed to `pitest-survival-check-xml.groovy` as CI runs both XML model type of suppression and previous HTML type of model (known as `pitest.sh`, now changed to `pitest-survival-check-html.sh`)

Reply:
Pr at https://github.com/checkstyle/checkstyle/pull/11763 for original pitest.sh script